### ISSUE TITLE
I would like to ask you some questions about the details of the paper

### ISSUE THREAD

**Issue by Bin-ze** [rahul-goel/isrf_code#5]
Thanks, very great work! I have some questions I would like to ask you about the details of the paper.
1. In Chapter 3.2, you described the optimization process of Semantic Features Distillation, and mentioned that we directly optimize for the features on the voxel grid in the TensoRF representation without a neural network, which means directly using the representation of tensoRF to calculate the difference with 2d features?
2. After the training is completed, use kmeans to generate the cluster center of the User Provided Stroke feature, and then use NNFM to get the feature with the highest confidence from the 4d field. Is there any problem with this understanding?
3. The Region Growing in chapter 3.4 is used for further optimization from the bitmap, but I don't understand what this optimization process is. Can you give some specific hints?
4. I am very interested in nerf scene editing. After getting a complete model, you described the centralized scene editing method in chapter 5.2, such as Affine transformation. The reason why the editing can be done here is that TensoRF has a clear 3d voxel representation ?

Looking forward to your reply! Thanks again

**Comment by rahul-goel** [comment]
Hello! Thank you for taking interest in our work.

1. There is no neural network involved in the prediction of semantic features by our method. We use a VM-64 TensoRF representation. Note that the collected features from the 3 different TensoRF VMs are combined through a learnable transformation matrix. And that is how we can get the semantic feature vector for any 3D point. The semantics at these 3D points are accumulated using volumetric rendering to get 2D semantics for a pixel. An L2 loss is applied here between the predicted 2D semantic and the one we obtain from DINO.

2. When a user makes a stroke, they select can scribble over a huge number of pixels. All the 2D features corresponding to these pixels are collected and clustered into 10 different clusters. NNFM is performed between the cluster centers of these 10 clusters and the entire 3D voxel grid.

3. I'm not sure what you mean by the optimization of the bitmap. We have a 3D bitmap that we obtain as the high confidence region. But this doesn't completely cover the desired object. We use region growing to to dilate this 3D bitmap so that it covers the object correctly and doesn't overshoot.

4. We are using a 3D voxel representation. But that is not the reason why editing is possible. For the methods that we are using for segmentation and editing, we find that voxels are a better representation. They also have a benefit of being fast to render. That being said, there has been quite a lot of editing work in neural network representation itself. For example, [http://editnerf.csail.mit.edu/](http://editnerf.csail.mit.edu/).

Do let me know if you have follow up questions. Else feel free to close the issue. Thanks.
